app/models/concerns/availability_status_concern.rb

```diff
@@ -19,13 +19,22 @@ module AvailabilityStatusConcern
   def update_status
     updated_attributes = changed - IGNORE_LIST
 
-    if updated_attributes.any?
-      self.availability_status = nil
-      self.last_checked_at = nil
+    remove_availability_status if updated_attributes.any?
+    remove_availability_status_on_source unless self.class == Source
+  end
+
+  def remove_availability_status
+    self.availability_status = nil
+    self.last_checked_at     = nil
+
+    if respond_to?(:availability_status_error)
+      self.availability_status_error = nil
+    end
+  end
 
-      if respond_to?(:availability_status_error)
-        self.availability_status_error = nil
-      end
+  def remove_availability_status_on_source
+    if availability_status.nil? || changed.include?(:availability_status)
```

Comment:
So this conditional is a bit odd to me - I think checking to see if the changes include availability status would be enough, since the `||` would fall through to the second conditional almost every time unless it was just created since `nil` is the default. 

If that's the reasoning there should at least be a comment explaining that, or we could have a default on this field since that is what it's doing.

Reply:
as I posted above, this condition will be never met for Authentication, because it's never set